api/src/main/java/org/openmrs/module/bedmanagement/dao/impl/BedManagementDaoImpl.java

```diff
@@ -243,4 +245,136 @@ public Bed getLatestBedByVisit(String visitUuid) {
                 .setParameter("voided", false)
                 .list();
     }
+
+    @Override
+    public List<AdmissionLocation> getAdmissionLocations() {
+        List<Integer> admissionLocationIds = getAdmissionLocationIds();
+        String sql = "select l from Location l " +
+                "where l.locationId in :admissionLocationIds and " +
+                "(l.parentLocation.locationId not in :admissionLocationIds or l.parentLocation.locationId is null) and " +
+                "l.retired=0";
+        Query query = sessionFactory.getCurrentSession().createQuery(sql);
+        query.setParameterList("admissionLocationIds", admissionLocationIds);
+        List<Location> locations = (List<Location>) query.list();
+
+        List<AdmissionLocation> admissionLocations = new ArrayList<AdmissionLocation>();
+        for(Location location : locations){
+            admissionLocations.add(this.getAdmissionLocationsByLocation(location));
+        }
+
+        return admissionLocations;
+    }
+
+    @Override
+    public AdmissionLocation getAdmissionLocationsByLocation(Location location) {
+        Session session = sessionFactory.getCurrentSession();
+        List<Integer> locationIds = new ArrayList<>();
+        locationIds.add(location.getId());
+        Set<Location> childLocations = location.getChildLocations();
+        if(childLocations != null)
+            for (Location childLocation : childLocations) {
+                locationIds.add(childLocation.getId());
```

Comment:
@sanishmaharjan, this loop is not necessary, you can use collections of objects in HQL queries:
```
"... where blm.location in (:locations)
 ..."
```
And then directly:
```
.setParameterList("locations", childLocations)
```

Reply:
So basically you can do something like:
```java
Set<Location> locations = new HashSet<Location>(Arrays.asList(location));
Set<Location> childLocations = location.getChildLocations();
if (!CollectionUtils.isEmpty(childLocations)) {
  locations.addAll(childLocations);
}

String hql = "select count(blm.bed) as totalBeds ," +
  " COALESCE(sum(CASE WHEN blm.bed IS NOT NULL AND blm.bed.status = :occupied THEN 1 ELSE 0 END), 0) as occupiedBeds" +
  " from BedLocationMapping blm where blm.location in (:locations)";

AdmissionLocation admissionLocation = (AdmissionLocation) session.createQuery(hql)
  .setParameterList("locations", locations)
  .setParameter("occupied", BedStatus.OCCUPIED.toString())
  .setResultTransformer(Transformers.aliasToBean(AdmissionLocation.class))
  .uniqueResult();
```